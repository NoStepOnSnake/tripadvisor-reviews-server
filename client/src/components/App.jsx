import React from 'react';
// import styled from 'styled-components';

// The htmlFor attribute of the <label> tag should be equal to the id attribute of the related 
// element to bind them together.

// Text
// rgb(0, 0, 0)
// font-family: 'Trip Sans', Arial, sans-serif !important
// font-size: 14px
// Number
// font-family: 'Trip Sans', Arial, sans-serif !important
// font-size: 14px
// rgb(118, 118, 118);

// const Button = styled.button`
//   background: red;
//   border-radius: 8px;
//   color: white;
// `;

// class App extends React.Component {
//   render() {
//     return (
//       <Button>Click Me</Button>
//     )
//   }
// }

// export default App;

class App extends React.Component {
  /**
   * Constructor
   * @param {Object} props - Short for properties; has all the values passed from parent component.
   */
  constructor(props) {
    super(props); // Sets `this.props`. Otherwise, when accessing `this.props`, would be `undefined`
    this.state = {
      selected: '',
    };

    this.handleInputRadioChange = this.handleInputRadioChange.bind(this);
  }

  /**
   * Handling Events: handle the change in input radio selection of languages.
   * @param {Object} target - The `target` property of the `Event` interface is a reference to the
   *                          object onto which the event was dispatched.
   * @param {String} name - The `name` attribute.
   * @param {String} selected - An alias for `name` parameter.
   */
  handleInputRadioChange({ target: { name: selected } }) {
    this.setState({
      selected,
    });
  }

  /**
   * Render component
   * @returns {JSX} A virtual DOM element
   */
  render() {
    const { selected } = this.state;
    return (
      <div>
        <p id="langHeader">Language</p>
        <label className="lang">
          <input
            type="radio"
            name="all" 
            value="All Languages"
            checked={selected === 'all'}
            onChange={this.handleInputRadioChange}
          />
          All Languages <span className="langCount">(99)</span>
        </label>
        <br />
        <label className="lang">
          <input
            type="radio"
            name="en"
            value="English"
            checked={selected === 'en'}
            onChange={this.handleInputRadioChange}
          />
          English
        </label>
        <br />
      </div>
    );
  }
}

export default App;
