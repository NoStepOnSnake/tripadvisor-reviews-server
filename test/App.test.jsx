import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../client/src/components/App';

Enzyme.configure({ adapter: new Adapter() });

describe('App', () => {
  test('has a valid snapshot', () => {
    const component = renderer.create(
      <App />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders two languages', () => {
    const element = shallow(
      <App />,
    );
    console.log('Hello World')
    console.log(element);
    expect(element.find('.language').length).toBe(0);
  });
});
