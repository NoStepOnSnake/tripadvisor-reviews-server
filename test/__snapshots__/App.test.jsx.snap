// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div>
  <p
    id="langHeader"
  >
    Language
  </p>
  <label
    className="lang"
  >
    <input
      checked={false}
      name="all"
      onChange={[Function]}
      type="radio"
      value="All Languages"
    />
    All Languages 
    <span
      className="langCount"
    >
      (99)
    </span>
  </label>
  <br />
  <label
    className="lang"
  >
    <input
      checked={false}
      name="en"
      onChange={[Function]}
      type="radio"
      value="English"
    />
    English
  </label>
  <br />
</div>
`;
